src/libraries/navdrawer/navdrawer.js

```diff
@@ -3,358 +3,369 @@
-                dragMode = 1;
-                newPos = deltaX - startPoint + options.width;
-                self.changeMenuPos();
-                const time = new Date().getTime() - (backgroundTouchStartTime || 0);
-                velocity = Math.abs(deltaX) / time;
+            if (deltaX < 0 && this.dragMode !== 2) {
+                this.dragMode = 1;
+                this.newPos = deltaX - this.startPoint + options.width;
+                this.changeMenuPos();
+                const time = new Date().getTime() - (this.backgroundTouchStartTime || 0);
+                this.velocity = Math.abs(deltaX) / time;
             }
         }
 
         e.preventDefault();
         e.stopPropagation();
     }
 
-    function onBackgroundTouchEnd(e) {
-        const touches = getTouches(e);
+    onBackgroundTouchEnd(e) {
+        const touches = this.getTouches(e);
         const touch = touches[0] || {};
         const endX = touch.clientX || 0;
-        const deltaX = endX - (backgroundTouchStartX || 0);
-        self.checkMenuState(deltaX);
-        countStart = 0;
+        const deltaX = endX - (this.backgroundTouchStartX || 0);
+        this.checkMenuState(deltaX);
+        this.countStart = 0;
     }
 
-    function onMaskTransitionEnd() {
-        const classList = mask.classList;
+    onMaskTransitionEnd() {
+        const classList = this.mask.classList;
 
         if (!classList.contains('backdrop')) {
             classList.add('hide');
         }
     }
 
-    let self;
-    let defaults;
-    let mask;
-    var newPos = 0;
-    var currentPos = 0;
-    var startPoint = 0;
-    var countStart = 0;
-    var velocity = 0;
-    options.target.classList.add('transition');
-    var dragMode = 0;
-    var scrollContainer = options.target.querySelector('.mainDrawer-scrollContainer');
-    scrollContainer.classList.add('scrollY');
-
-    const TouchMenuLA = function () {
-        self = this;
-        defaults = {
-            width: 260,
-            handleSize: 10,
-            disableMask: false,
-            maxMaskOpacity: 0.5
-        };
-        this.isVisible = false;
-        this.initialize();
-    };
+    initElements() {
+        const options = this.options;
 
-    TouchMenuLA.prototype.initElements = function () {
         options.target.classList.add('touch-menu-la');
         options.target.style.width = options.width + 'px';
         options.target.style.left = -options.width + 'px';
 
         if (!options.disableMask) {
-            mask = document.createElement('div');
-            mask.className = 'tmla-mask hide';
-            document.body.appendChild(mask);
-            dom.addEventListener(mask, dom.whichTransitionEvent(), onMaskTransitionEnd, {
+            this.mask = document.createElement('div');
+            this.mask.className = 'tmla-mask hide';
+            document.body.appendChild(this.mask);
+            dom.addEventListener(this.mask, dom.whichTransitionEvent(), this.onMaskTransitionEnd, {
```

Comment:
Need to bind (in the constructor) because `this` in the function will be `this.mask` (`div`) instead of navdrawer.